db/migrate/20221130184003_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```

Comment:
- [ ] according to project requirement kindly change `user` to `author` i.e `t.references :author, foreign_key: { to_table: 'users' }`